oauth2.go

```diff
@@ -149,6 +151,31 @@ func (c *Config) PasswordCredentialsToken(ctx context.Context, username, passwor
 	})
 }
 
+func (c *Config) getParams(base url.Values) url.Values {
+	var params url.Values
+	if c.Params != nil {
+		params = make(url.Values, len(*c.Params)+len(base))
+		for key, values := range *c.Params {
+			params[key] = make([]string, 0, len(values))
+			for _, v := range values {
+				params[key] = append(params[key], v)
+			}
+		}
+	} else {
+		params = make(url.Values, len(base))
+	}
+	for key, values := range base {
+		if _, ok := params[key]; ok {
+			continue
+		}
+		params[key] = make([]string, 0, len(values))
+		for _, v := range values {
+			params[key] = append(params[key], v)
+		}
```

Comment:
These four lines look the same as above, make a `cp` function inline inside this method?